test/parse-uri-components.test.js

```diff
@@ -1,12 +1,28 @@
 const Connection = require('../');
 const chai = require('chai');
 const expect = chai.expect;
+const proxyquire = require('proxyquire');
```

Comment:
Horray! Glad this worked out. Could you add a few lines of comments on why these stubs are needed? I imagine existing test suites could be enhanced with this fine grain control. Along with extracting codes paths that are only tested (if at all) via functional tests today finer grain to unit tests.

Reply:
Added comments. What do you think? Is this what you meant?